tests/bson/bson-objectid-getTimestamp-002.phpt

```diff
@@ -0,0 +1,31 @@
+--TEST--
+MongoDB\BSON\ObjectId::getTimestamp: Ensure that the Timestamp field is represented as an unsigned 32-bit integer
+--SKIPIF--
+<?php if (PHP_INT_SIZE !== 8) { die("skip Difficulty creating dates from unsigned 32-bit ints on a 32-bit platform"); } ?>
```

Comment:
Sorry for the previous suggestion, but let's change this to simply:

```
<?php if (8 !== PHP_INT_SIZE) { die('skip Only for 64-bit platform'); } ?>
```

I was poking around in the date extension's `parse_date.re` (which is used to generate the `scan()` function `parse_date.c` that processes the string argument for `date_create` and `DateTime::__construct()`) and it doesn't look like it has any problem working with large integers. The accepted pattern is `"@" "-"? [0-9]+;` and it stores the result in an unsigned long long, which would hold a 64-bit integer even on a 32-bit platform.

Provided we used `sprintf('%u', $ts)` below for both the echo and `date_create()` argument, I actually think the test would pass on both 32-bit and 64-bit platforms, but I'm happy to skip 32-bit for simplicity.

I just don't want to throw shade at @derickr with an inaccurate skip message :)

Reply:
Well, if we can easily have this pass on 32-bit I'm all for it.